src/Components/WebAssembly/WebAssembly/src/HotReload/WebAssemblyHotReload.cs

```diff
@@ -95,9 +97,87 @@ public static void ApplyHotReloadDelta(string moduleIdString, byte[] metadataDel
                     (metadataDelta, ilDeta)
                 };
             }
+        }
+
+        private static void ApplyUpdate(Assembly assembly, byte[] metadataDelta, byte[] ilDeta)
+        {
+            _handlerActions ??= GetMetadataUpdateHandlerActions();
+            var (beforeUpdates, afterUpdates) = _handlerActions.Value;
+
+            beforeUpdates.ForEach(a => a(null));
```

Comment:
So, if I understand correctly `beforeUpdates` and `afterUpdates` are handlers that can be invoked before and after deltas are applied. What are some scenarios where they are used?

Also -- why does this take a null value instead of just being a parameterless method?

Reply:
> What are some scenarios where they are used?

e.g. to clear corelib's reflection cache, to clear JsonSerializer's cache of reflection data, to refresh a binding in a UI, etc.

> why does this take a null value instead of just being a parameterless method?

The plan is for the agent to pass an array of the types being updated, if available, but we don't have that information yet.  The pattern will likely evolve over the next few months as we find out more about what various handlers need.